framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -171,15 +182,41 @@ public Void visitDeclared(AnnotatedDeclaredType type, Void p) {
          * If any constant-value annotation has &gt; MAX_VALUES number of values provided, treats
          * the value as UnknownVal. Works together with ValueVisitor.visitAnnotation, which issues a
          * warning to the user in this case.
+         *
+         * <p>Additional rules applied for IntVal and IntRange annotations: if anno is IntVal and
```

Comment:
Why is this whole class even necessary? If somebody wants to write a veeery long IntVal annotation, why stop them?
I think MAX_VALUES should only be used to stop very large values from arising from LUB and other computations.
Is there any harm in simply removing this class and whatever ValueVisitor.visitAnnotation does? (You haven't touched the latter, which is another bad sign.)